tools/clang/test/HLSL/ShaderOpArith.xml

```diff
@@ -581,6 +581,361 @@
+    <Resource Name="RTarget" Dimension="TEXTURE2D" Width="18" Height="18" Format="R32G32B32A32_FLOAT" Flags="ALLOW_RENDER_TARGET" InitialResourceState="COPY_DEST" />
+    <Resource Name="U0" Dimension="BUFFER" Width="11552"
+              Flags="ALLOW_UNORDERED_ACCESS" InitialResourceState="COPY_DEST"
+              Init="Zero" ReadBack="true" />
+    <Resource Name="U1" Dimension="BUFFER" Width="11552"
+              Flags="ALLOW_UNORDERED_ACCESS" InitialResourceState="COPY_DEST"
+              Init="Zero" ReadBack="true" />
+    <Resource Name="U2" Dimension="BUFFER" Width="11552"
+              Flags="ALLOW_UNORDERED_ACCESS" InitialResourceState="COPY_DEST"
+              Init="Zero" ReadBack="true" />
+
+    <RootValues>
+      <RootValue HeapName="ResHeap" />
+    </RootValues>
+    <DescriptorHeap Name="ResHeap" Type="CBV_SRV_UAV">
+      <Descriptor Name='T0' Kind='SRV' ResName='T0' />
+      <Descriptor Name='U0' Kind='UAV' ResName='U0'
+                  NumElements="722" StructureByteStride="16" />
+      <Descriptor Name='U1' Kind='UAV' ResName='U1'
+                  NumElements="722" StructureByteStride="16" />
+      <Descriptor Name='U2' Kind='UAV' ResName='U2'
+                  NumElements="722" StructureByteStride="16" />
+    </DescriptorHeap>
+    <DescriptorHeap Name="RtvHeap" NumDescriptors="1" Type="RTV">
+      <Descriptor Name="RTarget" Kind="RTV"/>
+    </DescriptorHeap>
+
+    <InputElements>
+      <InputElement SemanticName="POSITION" Format="R32G32B32_FLOAT" AlignedByteOffset="0" />
+      <InputElement SemanticName="TEXCOORD" Format="R32G32_FLOAT" AlignedByteOffset="12" />
+    </InputElements>
+    <RenderTargets>
+      <RenderTarget Name="RTarget"/>
+    </RenderTargets>
+    <Shader Name="CS" Target="cs_6_7" EntryPoint="CSMain" Text="@PS"/>
+    <Shader Name="VS" Target="vs_6_7" EntryPoint="VSMain" Text="@PS"/>
+    <Shader Name="MS" Target="ms_6_7" EntryPoint="MSMain" Text="@PS"/>
+    <Shader Name="AS" Target="as_6_7" EntryPoint="ASMain" Text="@PS"/>
+    <Shader Name="PS" Target="ps_6_7" EntryPoint="PSMain">
+      <![CDATA[
+        struct PSInput {
+          float4 position : SV_POSITION;
+          float2 uv : TEXCOORD;
+        };
+
+        Texture2D<float> g_tex : register(t0);
+        RWStructuredBuffer<uint4> g_bufMain : register(u0);
+        RWStructuredBuffer<uint4> g_bufMesh : register(u1);
+        RWStructuredBuffer<uint4> g_bufAmp : register(u2);
+
+        SamplerState g_samp : register(s0);
+        SamplerComparisonState g_sampCmp : register(s1);
+
+        int2 ClampOffset(int2 offset) {
+          if (offset.x < -8 || offset.x > 7)
+            offset.x = (offset.x << 28) >> 28;
+          if (offset.y < -8 || offset.y > 7)
+            offset.y = (offset.y << 28) >> 28;
+          return offset;
+        }
+
+        uint4 DoTests( int2 coord, int2 offset ) {
+#if __SHADER_TARGET_MINOR == 7
+          float cmp = (coord.y + offset.y)*1000 + coord.x + offset.x;
```

Comment:
`offset` should be clamped here so only the lower 4 bits are considered

Reply:
Nope. That's the point of this test, to verify that behavior when bounds are exceeded.

Reply:
sorry. I missed the context here. Yeah. the Clamp call is missing in the 6.7 variant

Reply:
Okay. Deleted my previous comments and I'm going to try again. 

The intent is that the clamping takes place by the implementation. Since the 6.6 code has no implementation, the test does the clamping. The results testing assumes that the clamping was performed either by the implementation or the test.

I considered performing the clamping within the compiler to make sure the limit was respected, but I didn't want to generate possibly unneeded code that would slow the operation down. So we have this HLK test to make sure the implementation performs the clamps.

Reply:
But this particular expression is computing a reference value without using texture ops. So there is no clamping that will be performed by the HW. As far as its concerned, it's just a standard 32-bit integer computation.

Reply:
I'm with you now